File: src/main/java/com/multicampus/gamesungcoding/a11ymarketserver/common/jwt/repository/RefreshTokenRepository.java
package com.multicampus.gamesungcoding.a11ymarketserver.common.jwt.repository;

import com.multicampus.gamesungcoding.a11ymarketserver.common.jwt.entity.RefreshToken;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;
import java.util.UUID;

public interface RefreshTokenRepository extends JpaRepository<RefreshToken, UUID> {

    Optional<RefreshToken> findByToken(String token);

    Optional<RefreshToken> findByUser_UserId(UUID userId);

    void deleteByToken(String token);
}

File: src/main/kotlin/com/multicampus/gamesungcoding/a11ymarketserver/common/jwt/repository/RefreshTokenRepository.kt
package com.multicampus.gamesungcoding.a11ymarketserver.common.jwt.repository

import com.multicampus.gamesungcoding.a11ymarketserver.common.jwt.entity.RefreshToken
import org.springframework.data.jpa.repository.JpaRepository
import java.util.*

interface RefreshTokenRepository : JpaRepository<RefreshToken, UUID> {
    fun findByToken(token: String): RefreshToken?

    fun findByUserUserId(userId: UUID): RefreshToken?

    fun deleteByToken(token: String)
}
